SIV_scraper

```diff
@@ -0,0 +1,215 @@
+setwd("L:/Prices/Dashboards/SIV")
+
+library(zoo)
+library(dplyr)
+library(readxl)
+library(xlsx)
+library(ggplot2)
+library(openxlsx)
+library(data.table)
+library(xml2)
+library(XML)
+library(purrr)
+library(readr)
+library(stringr)
+library(reshape2)
+library(data.table)
+library(pdftools)
+library(shiny)
+library(writexl)
+library(DT)
+library(tidyr)
+library(tibble)
+library(tabulizer)
+library(rvest)
+library(pdftools)
+library(shinythemes)
+
+filePath="https://ec.europa.eu/agriculture/fruit-and-vegetables"
+download.file(filePath, destfile = "Data/temp_files/scrapedpage.html", quiet=TRUE)
+
+ui <- fluidPage(theme = shinytheme("cosmo"),
+                
+                # Sidebar layout with a input and output definitions ----
+                navbarPage("SIV scraper",
+                           
+                           tabPanel("SIV data",
+                                    
+                                    # Output: Verbatim text for data summary ----
+                                    hr(),
+                                    actionButton("submitSIV", "Submit"),
+                                    downloadButton("dl_SIV", "Download"),
+                                    hr(),
+                                    DTOutput("SIV_view"))
+                                             
+                )#end of navbar page
+)#end of UI
+
+
+server <- function(input, output, session) {
+  
+  address=function(){
+    page <- read_html("Data/temp_files/scrapedpage.html")
+    page %>%
+      html_nodes("a") %>%       # find all links
+      html_attr("href") %>%     # get the url
+      str_subset("\\.pdf") %>% # find those that end in xlsx
+      .[[5]]                    # look at the first one
+
+  }
+
+  SIV_data=reactiveFileReader(
+    intervalMillis=86400000,
+    session=session,
+    filePath=address(),
+    readFunc = function(filePath) {
+       download.file(filePath, destfile="Data/temp_files/SIV.pdf", mode="wb")
+      data = pdf_data("Data/temp_files/SIV.pdf")
+      data <- data[[1]]
+      data
+    }
+  )
+  # SIV_data = reactive({
+  #   data = pdf_data("Data/temp_files/SIV.pdf")
+  #   data <- data[[1]]
+  #   data
+  # 
+  # })
+  ##Reactive to find data date
+  SIV_date = reactive({
+  SIV_date_row <- SIV_data()[grep("Date:", SIV_data()$text),4]
+  SIV_data_date <- SIV_data() %>% filter(y == as.numeric(SIV_date_row))
+  SIV_data_date$text <- as.Date(SIV_data_date$text, format = "%d/%m/%Y") 
+  SIV_data_date <- na.omit(SIV_data_date)
+  SIV_data_date <- SIV_data_date %>% dplyr::pull(text) 
+  SIV_data_date})
+  
+
+  SIV_data_clean = reactive({
+  ##Remove comment
+  #Remove starting rows
+  meta_row <- SIV_data()[grep("Product", SIV_data()$text), 4]
+  daily_data <- SIV_data() %>% filter(y >= as.numeric(meta_row))
+  
+  ##Clean data
+  find_cols <- daily_data %>% filter(text == "Product"| text == "Designation"| text == "Origin"| text == "Country"| text == "Market"| text == "Value"| text == "Quantity")  
```

Comment:
Could these multiple or statements be replaced by an in statement; see the %in% function